code/datums/gamemodes/gangwar.dm

```diff
@@ -1640,18 +2364,99 @@ proc/broadcast_to_all_gangs(var/message)
 		..()
 		return
 
-// /obj/item/chem_grenade/incendiary
-// 	name = "incendiary grenade"
-// 	desc = "A rather volatile grenade that creates a small fire."
-// 	icon = 'icons/obj/items/grenade.dmi'
-// 	icon_state = "incendiary"
-// 	icon_state_armed = "incendiary1"
-// 	stage = 2
-
-// 	New()
-// 		..()
-// 		var/obj/item/reagent_containers/glass/B1 = new(src)
-// 		B1.reagents.add_reagent("infernite", 20)
-// 		beakers += B1
-#undef CASH_DIVISOR
-#undef DEFAULT_MAX_GANG_SIZE
+// GANG TAGS
+
+/obj/decal/gangtag
+	name = "gang tag"
+	desc = "A gang tag, sprayed with nigh-uncleanable heavy metals."
+	density = FALSE
+	anchored = TRUE
+	layer = TAG_LAYER
+	icon = 'icons/obj/decals/graffiti.dmi'
+	icon_state = "gangtag0"
+	var/datum/gang/owners = null
+	var/list/mobs
+	var/heat = 0 // a rough estimation of how regularly this tag has people near it
+	var/image/heatTracker
+	var/active = TRUE
+	/// Deletes all duplicate tags (IE, from the same gang) on this tile
+	proc/delete_same_tags()
+		for(var/obj/decal/gangtag/T in get_turf(src))
```

Comment:
T means turf, don't use single letter variable names for things that aren't in ATOM generally.